catalog/resource_grants_test.go

```diff
@@ -600,6 +600,72 @@ func TestShareGrantUpdate(t *testing.T) {
 	}.ApplyNoError(t)
 }
 
+func TestGrantsForIdChange(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/unity-catalog/permissions/external_location/old.location?",
+				Response: catalog.PermissionsList{
+					PrivilegeAssignments: []catalog.PrivilegeAssignment{},
+				},
+			},
+			{
+				Method:   "PATCH",
+				Resource: "/api/2.1/unity-catalog/permissions/external_location/old.location",
+				ExpectedRequest: catalog.UpdatePermissions{
+					Changes: []catalog.PermissionsChange{
+						{
+							Principal: "me",
+							Add:       []catalog.Privilege{"CREATE_EXTERNAL_TABLE"},
+						},
+					},
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/unity-catalog/permissions/external_location/old.location?",
+				Response: catalog.PermissionsList{
+					PrivilegeAssignments: []catalog.PrivilegeAssignment{
+						{
+							Principal:  "me",
+							Privileges: []catalog.Privilege{"CREATE_EXTERNAL_TABLE"},
+						},
+					},
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/unity-catalog/permissions/external_location/new.location?",
+				Response: catalog.PermissionsList{
+					PrivilegeAssignments: []catalog.PrivilegeAssignment{
+						{
+							Principal:  "me",
+							Privileges: []catalog.Privilege{"CREATE_EXTERNAL_TABLE"},
+						},
+					},
+				},
+			},
+		},
+		Resource:    ResourceGrants(),
+		Update:      true,
```

Comment:
This is a very weird test. It verifies that the Update pathway works when the underlying external_location changes. However, in reality, the Update pathway is never taken when the external_location changes; instead, we delete and recreate. Obviously that can't be mocked with this test framework, but it is a bit weird.